src/msssim.jl

```diff
@@ -0,0 +1,85 @@
+"""
+Computes the MS-SSIM between two images.
+
+!!! info
+
+    MSSSIM, Multi-scale SSIM is defined only for gray images. RGB images are treated as 3d Gray
+    images. General `Color3` images are converted to RGB images first, you could
+    manually expand them using `channelview` if you don't want them converted to
+    RGB first.
+```
+
+# Reference
+
+[1] Wang, Z., Bovik, A. C., Sheikh, H. R., & Simoncelli, E. P. (2004). Image quality assessment: from error visibility to structural similarity. _IEEE transactions on image processing_, 13(4), 600-612.
+
+"""
+struct MSSSIM{N} <: FullReferenceIQI
+    kernel::AbstractArray{<:Real}
+    W::NTuple{N, <:Tuple} # number of scales inferred from length
+    function MSSSIM(kernel, W)
+        ndims(kernel) == 1 || throw(ArgumentError("only 1-d kernel is valid"))
+        issymetric(kernel) || @warn "MSSSIM kernel is assumed to be symmetric"
+        all(length.(W) .== 3) || throw(ArgumentError("(α, β, γ) required for all scales, instead it's $(W)"))
+        all([ all(x .>= 0) for x in W ]) || throw(ArgumentError("α, β, γ should be non-negative for all scales, instead it's $(W)"))
+        new{length(W)}(kernel, W)
+    end
+end
+
+# Weights for α, β, γ in [1]
+const MSSSIM_W = (0.0448, 0.2856, 0.3001, 0.2363, 0.1333) # α, β, γ for scales 1 through 5
+
+MSSSIM(kernel, W::NTuple{N, <:Real}) where N = MSSSIM(kernel, map(x->(x, x, x), W)) # shorthand for αᵢ=βᵢ=γᵢ for all scales i
```

Comment:
```suggestion
# shorthand for αᵢ=βᵢ=γᵢ for all scales i
# normalize γᵢ so that sum(γ) == 1 as the original paper suggested
MSSSIM(kernel, W::NTuple{N, <:Real}) where N = MSSSIM(kernel, map(x->(x, x, x), W ./ sum(W)))
```